### PR TITLE
chore(deps): add renovate.json

### PR DESCRIPTION
**Pull request by maxbrunet**
First attempt at a renovate config for parca-agent

The target branch is `renovate/configure`, so it is still a no-op to merge, Renovate will update its preview.

Documentation: https://docs.renovatebot.com/

### CODE REVIEW

**Review by maxbrunet — COMMENTED** [review]
Highlight and notes

File: renovate.json [hunk 1]
```diff
@@ -1,5 +1,115 @@
...
+    {
+      "description": "Group golang docker tags and rename to Golang",
+      "matchDatasources": ["docker"],
+      "matchPackageNames": ["golang"],
+      "matchPackagePatterns": ["/golang$"],
+      "commitMessageTopic": "Golang",
+      "groupName": "golang"
+    },
+    {
+      "description": "Group packages from kubernetes together",
```
Comment (kakkoyun) [inline renovate.json:56]:
```suggestion
      "description": "Group packages from Kubernetes together",
```
nit

**Review by kakkoyun — COMMENTED** [review]
LGTM.

I think we should just assign the PRs to @parca-dev/agent-maintainers otherwise :100: !

**Comment by kakkoyun** [comment]
Hmm, I would have expected this PR to be rebased against `main` when the `renovate/configure` is merged :thinking: 

@maxbrunet Could you please change the target branch for this PR?

**Comment by kakkoyun** [comment]
Don't fight with me renovate bot, I can click faster than you! 

> @maxbrunet Could you please change the target branch for this PR?

Ok, I got it now.

**Comment by maxbrunet** [comment]
> Hmm, I would have expected this PR to be rebased against `main` when the `renovate/configure` is merged thinking

I wanted these PR to be merged the other way around, so Renovate is fully configured when it ends up it starts working on `main`, but you make bold moves :grin:

File: renovate.json [hunk 2]
```diff
@@ -1,5 +1,116 @@
...
+    ":automergeMinor",
+    ":maintainLockFilesWeekly",
+    ":rebaseStalePrs",
+    ":semanticCommits",
+    ":semanticCommitScope(deps)",
+    "docker:pinDigests",
+    "helpers:pinGitHubActionDigests",
+    "regexManagers:dockerfileVersions"
+  ],
+  "assignees": ["kakkoyun"],
```
[the remark this answers is not in the document]
Comment (maxbrunet) [reply renovate.json:15]: Assignee does not support team, but we can try with reviewers: https://docs.renovatebot.com/configuration-options/#reviewers

And nearly forgot to mention, if your branch protection rule requires approval (or you would like it to), you can install this app: https://github.com/apps/renovate-approve

**Comment by kakkoyun** [comment]
> > Hmm, I would have expected this PR to be rebased against `main` when the `renovate/configure` is merged thinking
> 
> I wanted these PR to be merged the other way around, so Renovate is fully configured when it ends up it starts working on `main`, but you make bold moves grin

File: renovate.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (kakkoyun) [reply renovate.json]: Sounds good. We can always iterate if it doesn't fit our needs.

**Review by kakkoyun — APPROVED** [review]

**Pull request merged**